components/constellation/constellation.rs

```diff
@@ -2655,6 +2655,13 @@ where
                 },
             };
 
+        if let Some(old_pipeline) = self.pipelines.get(&old_pipeline_id) {
+            old_pipeline.notify_visibility(false);
+        }
+        if let Some(new_pipeline) = self.pipelines.get(&new_pipeline_id) {
+            new_pipeline.notify_visibility(true);
+        }
+
```

Comment:
I could have sworn we were already doing this. Am I going mad, and just dreaming up code? @cbrewster, can you remember?

Reply:
See #22986 - the visibility stuff is totally unused at the moment.

Reply:
It's been a while. I think we only updated activity, but not visibility.